test/telepilot_test.go

```diff
@@ -3,39 +3,55 @@ package telepilot_test
 import (
 	"context"
 	"crypto/tls"
-	"crypto/x509"
 	"errors"
+	"io"
 	"net"
 	"os"
 	"path"
+	"syscall"
 	"testing"
+	"time"
 
-	"github.com/google/uuid"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
-	"google.golang.org/grpc/status"
 
 	pb "go.creack.net/telepilot/api/v1"
 	"go.creack.net/telepilot/pkg/apiclient"
 	"go.creack.net/telepilot/pkg/apiserver"
 	"go.creack.net/telepilot/pkg/tlsconfig"
 )
 
+// Helper to assert success.
 func noError(t *testing.T, err error, msg string) {
 	t.Helper()
-	if err != nil {
+	// Ignore EOF/Closed Pipe errors as it is commonly used to notify closure.
+	if err != nil && !errors.Is(err, io.EOF) && !errors.Is(err, io.ErrClosedPipe) {
 		t.Fatalf("%s: %s.", msg, err)
 	}
 }
 
+// Generic assert.
 func assert[T comparable](t *testing.T, expect, got T, msg string) {
 	t.Helper()
 	if expect != got {
 		t.Fatalf("Assert fail: %s:\n Expect:\t%v\n Got:\t%v", msg, expect, got)
 	}
 }
 
+func assertChan(ctx context.Context, t *testing.T, expect string, ch <-chan []byte, msg string) {
```

Comment:
Maybe:

```suggestion
func assertChanOnce(ctx context.Context, t *testing.T, expect string, ch <-chan []byte, msg string) {
```

so it's clear that it's an assertion over a single receive?